packages/app-builder-lib/src/node-module-collector/nodeModulesCollector.ts

```diff
@@ -48,70 +85,78 @@ export abstract class NodeModulesCollector {
     }
   }
 
-  private convertToDependencyGraph(tree: DependencyTree): DependencyGraph {
-    const result: DependencyGraph = { ".": {} }
-
-    const flatten = (node: DependencyTree, parentKey = ".") => {
-      const dependencies = node.dependencies || {}
-
-      for (const [key, value] of Object.entries(dependencies)) {
-        // Skip empty dependencies(like some optionalDependencies)
-        if (Object.keys(value).length === 0) {
-          continue
-        }
-        const version = value.version || ""
-        const newKey = `${key}@${version}`
-        this.dependencyPathMap.set(newKey, path.normalize(this.resolvePath(value.path)))
-        if (!result[parentKey]?.dependencies) {
-          result[parentKey] = { dependencies: [] }
-        }
-        result[parentKey].dependencies!.push(newKey)
-
-        if (node.__circularDependencyDetected) {
-          continue
-        }
-        flatten(value, newKey)
+  private convertToDependencyGraph(tree: DependencyTree, parentKey = "."): DependencyGraph {
+    return Object.entries(tree.dependencies || {}).reduce<DependencyGraph>((acc, curr) => {
+      const [packageName, dependencies] = curr
+      // Skip empty dependencies (like some optionalDependencies)
+      if (Object.keys(dependencies).length === 0) {
+        return acc
+      }
+      const version = dependencies.version || ""
```

Comment:
If the version is empty, shouldn't we throw an error directly? It shouldn't be allowed to be empty, otherwise, it could lead to issues.

Reply:
We certainly can. This was already existing code, I just renamed `value`->`dependencies`. When you implemented this originally, was there a reason as to why we were allowing empty?

Reply:
Previously, in the type definition, `version` was optional, which is why this code was added. Now that `version` is no longer optional in the definition, this check is no longer needed.

Reply:
I can clean this up as part of any new issues that are reported since those require investigation as to why the dependency is missing a `path` as well.